miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/RealMiniApp.kt

```diff
@@ -45,6 +55,9 @@ internal class RealMiniApp(
         queryParams: String
     ): MiniAppDisplay = when {
         appId.isBlank() -> throw sdkExceptionForInvalidArguments()
+        miniAppManifestCache.isRequiredPermissionDenied(
```

Comment:
On iOS, it's first downloading the mini app and then checking if permissions are denied. It's a subtle distinction, but basically the downloading of the mini app shouldn't be dependent on whether the permissions are granted. It's the launching of the mini app that is dependent on the permissions.

So basically, I think you should move the call to `isRequestedPermissionsDenied` to after `miniAppDownloader.getMiniApp`.